app/components/app_patient_session_record_component.rb

```diff
@@ -21,7 +21,8 @@ def render?
     patient.consent_given_and_safe_to_vaccinate?(programme:) &&
       (
         patient_session.registration_status&.attending? ||
-          patient_session.registration_status&.completed? || false
+          patient_session.registration_status&.completed? ||
+          !patient_session.session.requires_registration?
```

Comment:
The logic was wrong initially as I didn't fully understand what the criteria was saying. Basically, if the session doesn't require registration, then the user should still be able to record a vaccination.

I also got rid of the `false`. I don't think it's needed as the guard operated `&` will give `nil` which is falsey anyway.